### DISCUSSION TITLE
setAnchor causes all nodes and their children to get enabled by default?

### DISCUSSION THREAD

**Discussion by sarimmehdi** [sceneview/sceneform-android#318] [Q&A]
Hello, in my current application, I sometimes disable some nodes that are children of an anchor node. The anchor, belonging to the anchor node, can then be hosted using the following code:

```
Anchor hostingAnchor = arCoreSession.hostCloudAnchor(anchor);
anchorNode.setAnchor(hostingAnchor);
```
I notice that the setAnchor method causes not just the anchor node but all its children to get enabled by default which means that the items the user disables in my app, will show up all of a sudden when the anchor starts hosting. The way that I dealt with it was to check whether certain nodes were set to disable or not by the user and then perform the necessary action if the node was enabled when it was supposed to be disabled.

However, I was wondering what exactly is the intuition behind enabling the anchor node and its children whenever the setAnchor method is called with the anchor node as its argument. For reference, I am linking to the relevant [code segment](https://github.com/SceneView/sceneform-android/blob/ba2c818151b91296f048711f494e1acc239ff809/core/src/main/java/com/google/ar/sceneform/AnchorNode.java#L50) from Sceneform.

**Comment by grassydragon** [comment]
Hi!
I think this is a drawback of the `isTracking()` method because we also need to check the cloud anchor state: https://developers.google.com/ar/reference/java/com/google/ar/core/Anchor#getCloudAnchorState()
I usually create the `AnchorNode` after the cloud anchor is resolved. So I have never experienced this problem.

**Comment by ThomasGorisse** [comment]
>However, I was wondering what exactly is the intuition behind enabling the anchor node and its children whenever the setAnchor method is called with the anchor node as its argument. For reference, I am linking to the relevant code segment from Sceneform.

What is sure is that we removed this behavior from SceneView-android. But I can't remember why it was there.
Couldn't you make your nodes invisible instead of disable?

**Reply by sarimmehdi** [reply]
I do not want the user to be able to interact with them at all. I believe turning nodes invisible doesn't prevent that and the user might still accidentally touch the screen where the node is. Also, disabling nodes saves memory right?

**Reply by ThomasGorisse** [reply]
Yep. You need to override the Node and don't call super on `setEnable()`
